packager/media/formats/mp2t/ts_segmenter.cc

```diff
@@ -171,30 +171,46 @@ Status TsSegmenter::WritePesPacketsToFile() {
 Status TsSegmenter::FinalizeSegment(uint64_t start_timestamp,
                                     uint64_t duration) {
   if (!pes_packet_generator_->Flush()) {
-    return Status(error::MUXER_FAILURE,
-                  "Failed to flush PesPacketGenerator.");
+    return Status(error::MUXER_FAILURE, "Failed to flush PesPacketGenerator.");
   }
   Status status = WritePesPacketsToFile();
   if (!status.ok())
     return status;
 
-  // This method may be called from Finalize() so ts_writer_file_opened_ could
+  // This method may be called from Finalize() so start_of_new_segment_ could
   // be false.
-  if (ts_writer_file_opened_) {
-    if (!ts_writer_->FinalizeSegment()) {
-      return Status(error::MUXER_FAILURE, "Failed to finalize TsWriter.");
+  if (start_of_new_segment_) {
```

Comment:
Actually can we call the variable: `segment_started_`, which seems better than `start_of_new_segment_`. Sorry for changing the name multiple times.

Also, I think we can improve the readability here by return early if it is not the case:

```c++
  // This method may be called from Finalize() so start_of_new_segment_ could
  // be false.
  if (!segment_started_)
    return Status::OK;

  std::string segment_path ...
```